src/matcher/gui/IGuiComponent.java

```diff
@@ -22,4 +22,14 @@ default void onFieldSelect(FieldInstance field) { }
 	default void onMethodVarSelect(MethodVarInstance arg) { }
 
 	default void onMatchListRefresh() { }
+
+	enum ViewChangeCause {
+		SORTING_CHANGED,
+		CLASS_TREE_VIEW_TOGGLED,
+		SHOW_NON_INPUTS_TOGGLED,
+		THEME_CHANGED,
+		DIFF_COLORS_TOGGLED,
+		NAME_TYPE_CHANGED,
+		DECOMPILER_CHANGED
+	}
```

Comment:
Maybe it would be better to turn these into proper objects, which can inherit from each other? So that plugins can define their own `ViewChangeCause`s, without breaking the GUI

Reply:
To be fair, that's out of scope for this PR. Once the Core/GUI split happens, we'll need to refactor the plugin API anyway